gobblin-runtime/src/main/java/gobblin/runtime/spec_store/FSSpecStore.java

```diff
@@ -194,32 +207,44 @@ public Spec getSpec(URI specUri, String version) throws IOException, SpecNotFoun
   public Collection<Spec> getAllVersionsOfSpec(URI specUri) throws IOException, SpecNotFoundException {
     Preconditions.checkArgument(null != specUri, "Spec URI should not be null");
 
-    Collection<Spec> specs = Lists.newArrayList();
-    FileStatus[] fileStatuses = fs.listStatus(this.fsSpecStoreDirPath);
-    for (FileStatus fileStatus : fileStatuses) {
-      if (StringUtils.startsWith(fileStatus.getPath().getName(), specUri.toString())) {
-        specs.add(readSpecFromFile(fileStatus.getPath()));
+    Collection<Spec> specs = getSpecs();
```

Comment:
We should be able to pass down the filtering here to avoid listing directories that won't match.